src/master.rs

```diff
@@ -517,9 +517,101 @@ impl Master {
         Ok(time)
     }
 
+    pub fn foe_read(&mut self, idx: SlavePos, name: &str) -> Result<Vec<u8>> {
+        let file_name = string_to_foe_name(name)?;
+        // FIXME: this is the same as in the c-implementation. Should read in chunks instead of a
+        // fixed size buffer. The ioctl-call in the master pre-allocates a 10000 byte buffer, so we
+        // do the same here.
+        const FOE_SIZE: usize = 10_000;
+        let mut buf: Vec<u8> = vec![0; FOE_SIZE];
+        let mut data = ec::ec_ioctl_slave_foe_t {
+            slave_position: idx.into(),
+            offset: 0,
+            buffer_size: FOE_SIZE as u64,
+            buffer: buf.as_mut_ptr(),
+            file_name,
+            ..Default::default()
+        };
+        ioctl!(self, ec::ioctl::SLAVE_FOE_READ, &mut data)?;
+
+        assert!(data.data_size <= FOE_SIZE as u64);
+        buf.resize(data.data_size as usize, 0x0);
```

Comment:
Nit: since this never increases size, maybe `truncate` is semantically better.

Reply:
Good idea, wasn't aware of the difference